### PR TITLE
More introductory material

### PR DESCRIPTION
**Pull request by gpatkinson**
Great website. I thought that some people would benefit from more introductory words on bitcoin script so added a few more pages to that section. I also made a few minor organizational changes that just made sense to me. Thanks for considering.

### CODE REVIEW

File: script/tx_validation.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Transaction Validation
+
+Bitcoin validation relies on two general types of scripts:
+1. **locking script** - the criteria that must be met in order to spend a UTXO
+2. **unlocking script** - the script that satisfies the locking script, typically a cryptographic signagure
+    
+These script types are used together in transaction validation:
+
+>Every bitcoin validating node will validate transactions by executing the locking and unlocking scripts together. Each input contains an unlocking script and refers to a previously existing UTXO. The validation software will copy the unlocking script, retrieve the UTXO referenced by the input, and copy the locking script from that UTXO. The unlocking and locking script are then executed in sequence. The input is valid if the unlocking script satisfies the locking script conditions ... All the inputs are validated independently, as part of the overall validation of the transaction.
```
Comment (vostrnad) [inline script/tx_validation.md]: The terminology "locking script" and "unlocking script" is not recommended anymore, according to Murch's proposed [terminology BIP](https://github.com/murchandamus/bips/pull/1) (see "Locking Script and Unlocking Script").

**Comment by thunderbiscuit** [comment]
Hey thanks for the kind words! This PR will now need a rebase, and I left some comments below. Thanks for contributing!

File: .vitepress/config.js [hunk 2]
```diff
@@ -9,8 +9,9 @@ export default defineConfig({
     },
 
     nav: [
+      { text: "Intro", link: "/script/", activeMatch: "/script/" },
```
Comment (thunderbiscuit) [inline .vitepress/config.js]: At the moment I don't intend on changing the name of the navbar navigation item to "Intro" (not that I'm against new names or even new sections entirely). I'm just not sure intro quite captures what I have in mind for this section. I also want to keep it in second position, as the primary goal of the website (at least for now) is to provide a complete encyclopedia of opcodes and short pages exploring them.

File: .vitepress/config.js [hunk 3]
```diff
@@ -290,6 +291,18 @@ export default defineConfig({
       ],
       
       "/script/": [
+     //  COMMENT: OPTIONAL 'SECTIONING' FORMAT OF INTRODUCTORY PAGES
```
Comment (thunderbiscuit) [inline .vitepress/config.js]: The idea of breaking those pages into sections is definitely something I'm interested in for the future. You can remove the commented out lines for now, but as we add more pages I do think it's somewhere we want to go.

File: index.md [hunk 4]
```diff
@@ -14,13 +14,14 @@ hero:
   #     link: /overview/
 
 features:
-  - title: All opcodes
-    details: A small encyclopedia of all current opcodes.
+  - title: Introduction
```
Comment (thunderbiscuit) [inline index.md]: - I want to keep the opcode link as the first one.
- I don't think it's only introductory material in there!

File: script/index.md [hunk 5]
```diff
@@ -1,2 +1,2 @@
 # Bitcoin Script
-The pages in this section explore some of the more meta parts of the bitcoin scripting system.
+The pages in this section explore the bitcoin scripting system.
```
Comment (thunderbiscuit) [inline script/index.md:2]: Good simplification. Thanks!

File: script/script_basics.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Script Basics
```
Comment (thunderbiscuit) [inline script/script_basics.md:1]: Great idea to add an intro page on bitcoin script. I was jumping into the deep end a bit too quickly there.

File: script/the_stack.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# The Stack
```
Comment (thunderbiscuit) [inline script/the_stack.md]: Great page. I've been meaning to add a page on the [alternate stack](https://github.com/thunderbiscuit/opcode-explained/issues/1), and clearly a page on the stack itself should be there.

File: script/the_stack.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
+Script's stack allows for only two operations: push and pop. 
+- push adds something to the top of the stack
+- pop removes an item from the top of the stack. 
+
+As you may have noticed, only the top item from the stack can be manipulated.
+
+All of the opcodes listed on this website are specific variants of these two operations. For example, [OP_ADD](/opcodes/OP_ADD.md) pops two consecutive items from the stack, adds them together, and then pushes the result to the stack. Opcodes are the fundamental building blocks of transactions and, in various combinations, represent the full expressivity of the Bitcoin network. From time-to-time, [new opcodes are proposed](https://github.com/jamesob/bips/blob/jamesob-23-02-opvault/bip-0345.mediawiki) to give Bitcoin more capabilities.
```
Comment (thunderbiscuit) [inline script/the_stack.md]: I like the idea of pointing out that new opcodes are proposed and that opcodes in general are not set in stone. In this case let's remove the link to a particular proposal however.

File: script/tx_validation.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# Transaction Validation
```
Comment (thunderbiscuit) [inline script/tx_validation.md]: I'm not as sure about this page as I am the others. I wonder if adding some of this to the "Script Success and Failures" would make sense? There is also the issue of the new nomenclature for locking/unlocking scripts as mentioned by @vostrnad.

Overall I feel like this page might be better left for another PR where we can iterate on it there. Do you mind removing it from this one and we can maybe iterate over the idea in a different PR?

**Review by thunderbiscuit — CHANGES_REQUESTED** [review]
Hey thanks for the hard work! I left some comments. Happy to see people interested in the project enough to spend time contributing to it. Cheers!

File: script/tx_validation.md (hunk 1 above)
Comment (gpatkinson) [reply script/tx_validation.md]: Yeah, good point. I'll update it.

File: index.md (hunk 4 above)
Comment (gpatkinson) [reply index.md]: done and fair point!

File: .vitepress/config.js (hunk 2 above)
Comment (gpatkinson) [reply .vitepress/config.js]: sounds good, I'll revert

File: script/tx_validation.md (hunk 9 above)
Comment (gpatkinson) [reply script/tx_validation.md]: Sure. I agree it needs some work, including the updating the terminology.

**Review by gpatkinson — COMMENTED** [review]
I've made all the requested changes, thanks for your feedback.

File: script/script_basics.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+The Bitcoin ledger consists of a large, dynamic set of Unspent Transaction Outputs (UTXO's) monitored by a world-wide network of nodes. Control of these outputs is dictated by private keys, which can cryptographically sign transaction messages. A signature proves the ownership of a private key without revealing it to the network. Valid signatures result in the reassignment of UTXOs to other wallet addresses, a process also known as "spending."
```
Comment (vostrnad) [inline script/script_basics.md]:
```suggestion
The Bitcoin ledger consists of a large, dynamic set of Unspent Transaction Outputs (UTXOs) monitored by a world-wide network of nodes. Control of these outputs is dictated by private keys, which can cryptographically sign transactions. A signature proves the ownership of a private key without revealing it to the network. Valid signatures result in the reassignment of UTXOs to other owners, a process also known as "spending". 
```

File: script/script_basics.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# Script Basics
+
+The Bitcoin ledger consists of a large, dynamic set of Unspent Transaction Outputs (UTXO's) monitored by a world-wide network of nodes. Control of these outputs is dictated by private keys, which can cryptographically sign transaction messages. A signature proves the ownership of a private key without revealing it to the network. Valid signatures result in the reassignment of UTXOs to other wallet addresses, a process also known as "spending." 
+
+The language through which the network participants communicate is simply called 'Script' or 'Bitcoin Script.' Script is unlike modern programming languages--it most closely resembles a stack-oriented language from the late 1960's called [Forth](https://en.wikipedia.org/wiki/Forth_(programming_language)). Satoshi chose this style for its simplicity and minimal storage requirements.
```
Comment (vostrnad) [inline script/script_basics.md]:
```suggestion
The language through which the network participants communicate is simply called 'Script' or 'Bitcoin Script.' Script is unlike modern programming languages–it most closely resembles a stack-oriented language from the late 1960's called [Forth](https://en.wikipedia.org/wiki/Forth_(programming_language)). Satoshi chose this style for its simplicity and minimal storage requirements. 
```

File: script/script_basics.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
+# Script Basics
+
+The Bitcoin ledger consists of a large, dynamic set of Unspent Transaction Outputs (UTXO's) monitored by a world-wide network of nodes. Control of these outputs is dictated by private keys, which can cryptographically sign transaction messages. A signature proves the ownership of a private key without revealing it to the network. Valid signatures result in the reassignment of UTXOs to other wallet addresses, a process also known as "spending." 
+
+The language through which the network participants communicate is simply called 'Script' or 'Bitcoin Script.' Script is unlike modern programming languages--it most closely resembles a stack-oriented language from the late 1960's called [Forth](https://en.wikipedia.org/wiki/Forth_(programming_language)). Satoshi chose this style for its simplicity and minimal storage requirements. 
+
+Some other important elements of Script:
```
Comment (vostrnad) [inline script/script_basics.md]:
```suggestion
Some other important properties of Script:
```

File: script/script_basics.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+
+- NOT Turing Complete - this prevents loop-based Denial-of-Service attacks on the network
```
Comment (vostrnad) [inline script/script_basics.md]:
```suggestion
- Turing incomplete – this prevents loop-based denial-of-service attacks on the network
```

File: script/script_basics.md [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+- Stateless - any valid construction contains all necessary information for network participants to validate it
```
Comment (vostrnad) [inline script/script_basics.md]:
```suggestion
- Stateless – any valid construction contains all necessary information for network participants to validate it
```

File: script/the_stack.md [hunk 15]
```diff
@@ -0,0 +1,9 @@
+# The Stack
+
+Script's stack allows for only two operations: push and pop. 
+- push adds something to the top of the stack
+- pop removes an item from the top of the stack. 
+
+As you may have noticed, only the top item from the stack can be manipulated.
+
+All of the opcodes listed on this website are specific variants of these two operations. For example, [OP_ADD](/opcodes/OP_ADD.md) pops two consecutive items from the stack, adds them together, and then pushes the result to the stack. Opcodes are the fundamental building blocks of transactions and, in various combinations, represent the full expressivity of the Bitcoin network. From time-to-time, new opcodes are proposed to give Bitcoin more capabilities.
```
Comment (vostrnad) [inline script/the_stack.md]: This isn't true. Many opcodes manipulate multiple stack items, and some (like `OP_ROLL`) can manipulate a stack item at arbitrary depth. The last paragraph also contains information completely unrelated to the stack.

I'm not sure we even need a dedicated page for the stack, so I'd remove this for now and revisit later.

Comment (gpatkinson) [reply script/the_stack.md]: Thanks for the correction. As someone newer to understanding opcodes, I personally think a page on "the stack"would be helpful to others who are also interested but less experienced. I'm happy to remove and re-work for a possible future PR if that's the preferred route @thunderbiscuit

Comment (gpatkinson) [reply script/the_stack.md]: The more I consider it, I agree that it would make sense to re-work this for a separate PR. I've removed it for now.

**Review by gpatkinson — COMMENTED** [review]
All suggestions implemented.

File: index.md [hunk 16]
```diff
@@ -14,14 +14,16 @@ hero:
...
-
-  - title: Learn more about bitcoin script
-    details: More documentation on the nitty gritty of the bitcoin scripting language.
+  - title: Script
+    details: The basics of the bitcoin scripting language.
     link: /script/
 
+
+
+
```
Comment (vostrnad) [inline index.md]: nit: unnecessary changes to whitespace (especially at the end)

File: script/script_basics.md [hunk 17]
```diff
@@ -0,0 +1,14 @@
+# Script Basics
+
+The Bitcoin ledger consists of a large, dynamic set of Unspent Transaction Outputs (UTXOs) monitored by a world-wide network of nodes. Control of these outputs is dictated by private keys, which can cryptographically sign transactions. A signature proves the ownership of a private key without revealing it to the network. Valid signatures result in the reassignment of UTXOs to other owners, a process also known as "spending". 
+
+The language through which the network participants communicate is simply called 'Script' or 'Bitcoin Script.' Script is unlike modern programming languages–it most closely resembles a stack-oriented language from the late 1960's called [Forth](https://en.wikipedia.org/wiki/Forth_(programming_language)). Satoshi chose this style for its simplicity and minimal storage requirements.
```
Comment (vostrnad) [inline script/script_basics.md]:
```suggestion
The language through which the network participants communicate is simply called Script or Bitcoin Script. Script is unlike modern programming languages–it most closely resembles a stack-oriented language from the late 1960's called [Forth](https://en.wikipedia.org/wiki/Forth_(programming_language)). Satoshi chose this style for its simplicity and minimal storage requirements. 
```

File: script/script_basics.md [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+The language through which the network participants communicate is simply called 'Script' or 'Bitcoin Script.' Script is unlike modern programming languages–it most closely resembles a stack-oriented language from the late 1960's called [Forth](https://en.wikipedia.org/wiki/Forth_(programming_language)). Satoshi chose this style for its simplicity and minimal storage requirements. 
+
+Some other important properties of Script:
+
+- Turing incomplete – this prevents loop-based denial-of-service attacks on the network
+
+- Stateless – any valid construction contains all necessary information for network participants to validate it
+   
+
+
```
Comment (vostrnad) [inline script/script_basics.md]: nit: one newline at the end would be enough

File: typescript [hunk 19]
```diff
@@ -0,0 +1,80 @@
+Script started on 2023-11-26 17:51:23-05:00 [TERM="xterm-256color" TTY="/dev/pts/0" COLUMNS="80" LINES="24"]
```
Comment (vostrnad) [inline typescript]: I don't think you wanted to commit this?

**Review by vostrnad — CHANGES_REQUESTED** [review]

File: script/script_basics.md (hunk 18 above)
Comment (gpatkinson) [reply script/script_basics.md]: corrected, thanks.

Comment (vostrnad) [reply script/script_basics.md]: Not corrected, as far as I can tell.

Comment (gpatkinson) [reply script/script_basics.md]: my bad. all set.

**Review by vostrnad — APPROVED** [review]

**Review by gpatkinson — COMMENTED** [review]
All set. Ended up removing tx_validation.md for now per @vostrnad recs, will try again in a future PR

**Review by gpatkinson — COMMENTED** [review]
completed.

**Review by gpatkinson — COMMENTED** [review]
removed.

**Review by thunderbiscuit — CHANGES_REQUESTED** [review]
Thanks for the hard work! This is getting ready to merge, but I'd need a squash of all these commits into one (maybe two if you really wanted to separate the small syntax fixes from the new page addition).

Would you like to do this yourself? Otherwise I'm happy to do it and merge today.

**Comment by gpatkinson** [comment]
Thank you! Honored to be a part of this. As I'm sure you've noticed, I'm pretty new to GitHub collaboration (though I've used Git for a while). I'm happy to try this commit compression but I feel like I'm usually just stumbling around this GUI--if it's not too much trouble, I'd appreciate your help to make sure I don't botch something and further slow the merge. :)

**Comment by thunderbiscuit** [comment]
Hey no problem! Always happy to help. In this case what I'll be using is
1. an interactive rebase to squash all your commits into one. If you're using git from the command line this is the article I like to refer people to: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history
2. a rebase of that commit onto master

Just a tip: one thing that might give you trouble now is that you have made these commits on your master branch, but I'm rewriting history and merging that to mine. So now your master is off. An easy way to avoid that is to develop your PRs on new branches that don't affect your master.

Cheers!

**Comment by gpatkinson** [comment]
Thanks, I really appreciate it
